src/client.js

```diff
@@ -0,0 +1,97 @@
+"use strict";
+var W3CWebSocket = require('websocket').w3cwebsocket;
+class Client {
+  constructor(url, protocol) {
+    this.url = url;
+    this.protocol = protocol;
+    this.client = new W3CWebSocket(url, protocol);
+    this.subscriptions = {}; // id: handler
+    this.max_id = 0;
+
+    this.client.onopen = () => {
+      console.log("Websocket client connected.");
+    };
+
+    this.client.onmessage = (message) => {
+      let message_data = JSON.parse(message.data);
+      if (message_data.type === 'subscription_fail') {
```

Comment:
use constants for 'subscription_fail', 'subscription_data', etc, and import them in server and client.